src/engine/SpatialJoinAlgorithms.cpp

```diff
@@ -149,35 +155,32 @@ Id SpatialJoinAlgorithms::computeDist(const IdTable* idTableLeft,
                                       const IdTable* idTableRight,
                                       size_t rowLeft, size_t rowRight,
                                       ColumnIndex leftPointCol,
-                                      ColumnIndex rightPointCol) const {
-  auto getAreaOrPointGeometry = [&](const IdTable* idtable, size_t row,
-                                    size_t col, std::optional<GeoPoint> point) {
-    std::optional<AnyGeometry> geometry;
-
+                                      ColumnIndex rightPointCol) {
+  auto getAreaOrPointGeometry =
+      [&](const IdTable* idtable, size_t row, size_t col,
+          std::optional<GeoPoint> point) -> std::optional<size_t> {
     if (!point) {
       // nothing to do. When parsing a point or an area fails, a warning
       // message gets printed at another place and the point/area just gets
       // skipped
-      geometry = getAnyGeometry(idtable, row, col);
-    } else {
-      geometry = convertGeoPointToPoint(point.value());
+      return getAnyGeometry(idtable, row, col);
```

Comment:
The comment above this line seems unrelated and out of sync now, it can be removed.